packages/engine/crawlers/solana/transactionHistory.ts

```diff
@@ -0,0 +1,530 @@
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import type {
+	ConfirmedSignatureInfo,
+	Connection,
+	ParsedInstruction,
+	ParsedTransactionWithMeta,
+	PartiallyDecodedInstruction,
+} from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import type { Collectible, Token, TokenAccount } from '@walless/core';
+import { Networks } from '@walless/core';
+
+import { historyActions } from './../../state/history/index';
+import { getMetadata, solMint } from './metadata';
+import type { SolanaContext } from './shared';
+
+const gasilonConfigs = await fetch(`${GASILON_ENDPOINT}`, {
+	method: 'GET',
+	headers: {
+		'Content-Type': 'application/json',
+	},
+});
+
+const FEE_PAYER = (await gasilonConfigs.json()).feePayer;
+export interface Transaction {
+	id: string;
+	signature: string;
+	network: Networks;
+	type: 'sent' | 'received';
+	status: 'success' | 'pending' | 'failed';
+	sender: string;
+	receiver: string;
+	token: Token | Omit<Collectible, 'account' | 'collectionId'>;
+	fee: number;
+	tokenForFee?: Token;
+	preBalance?: number;
+	postBalance?: number;
+	amount: number;
+	date: Date;
+}
```

Comment:
I have concern about this `Transaction` type, a transaction on solana might include multiple transfers or multiple changes to accounts. So just using a token in a transaction is enough to simulate history?

Reply:
what 'just using a token' you mean?